scripts/tests/equipment/equipment_weapons.lua

```diff
@@ -0,0 +1,82 @@
+-----------------------------------
+-- Equipment Weapon TP Reset Tests
+-- Tests that TP is reset when equipping/unequipping weapons from various slots
+--
+-- Note: slotID vs equipSlotID distinction:
+--   - slotID: Inventory slot ID (0 = "unequip operation")
+--   - equipSlotID: Equipment slot being modified (SLOT_MAIN=0, SLOT_SUB=1, SLOT_RANGED=2, etc.)
+--
+-- Unequip flow in EquipItem():
+--   if (slotID == 0) → UnequipItem(equipSlotID) → TP reset handled in UnequipItem() line 2141
+-----------------------------------
+
+describe('Equipment Weapons - TP Reset on Unequip', function()
+    ---@type CClientEntityPair
+    local player
+
+    before_each(function()
+        -- Spawn NIN/10 with Dual Wield unlocked for full equipment testing
+        player = xi.test.world:spawnPlayer({
+            level = 10,
+            mjob = xi.job.NIN,
+            zone = xi.zone.WEST_RONFAURE
+        })
+    end)
+
+    it('should reset TP when unequipping main hand weapon', function()
+        -- Arrange: Equip main hand weapon and set TP
+        player:addItem(xi.item.BRONZE_DAGGER)
+        player:equipItem(xi.item.BRONZE_DAGGER, nil, xi.slot.MAIN)
+        player:setTP(2500)
+
+        -- Act: Unequip main hand weapon (slotID = 0)
```

Comment:
The slot ID being 0 under the hood is entirely irrelevant to this test? Everything is wrapped up in human-readable enums, etc.

Reply:
Removed comment and changed charutils to use enum value